### PR TITLE
Add support for oneof_ref

### PR DESCRIPTION
**Pull request by gmlewis**
Fixes: #68.

### CODE REVIEW

**Comment by andreafalzetti** [comment]
It'd be great to keep this pkg alive, I am happy to help review PRs cc @samlown

File: examples_test.go [hunk 1]
```diff
@@ -16,6 +16,8 @@ type SampleUser struct {
 	BirthDate   time.Time              `json:"birth_date,omitempty" jsonschema:"oneof_required=date"`
 	YearOfBirth string                 `json:"year_of_birth,omitempty" jsonschema:"oneof_required=year"`
 	Metadata    interface{}            `json:"metadata,omitempty" jsonschema:"oneof_type=string;array"`
+	IPAddress   interface{}            `json:"ipAddress,omitempty" jsonschema:"oneof_ref=#/$defs/ipv4;#/$defs/ipv6"`
+	IPAddresses []interface{}          `json:"ipAddresses,omitempty" jsonschema:"oneof_ref=#/$defs/ipv4;#/$defs/ipv6"`
```
Comment (samlown) [inline examples_test.go:20]: I appreciate these are only samples, but I'd still stick to the common pattern of `ip_address`.

**Review by samlown — APPROVED** [review]
Looks good! Thanks! I now handle these types of situations with a `JSONSchemaExtend` method on the model, but I can see how this is more convenient.

File: examples_test.go (hunk 1 above)
Comment (gmlewis) [reply examples_test.go]: Sorry for the delay.  Done.  PTAL.

**Pull request merged**